core/base/connectedComponents/ConnectedComponents.h

```diff
@@ -0,0 +1,159 @@
+                         std::vector<TID> &stack,
+
+                         const TT *triangulation,
+                         const TID seed) const {
+      // get component id
+      const TID componentId = components.size();
+
+      int stackSize = 1;
+      stack[0] = seed;
+      labels[seed] = componentId;
+
+      TID cIndex;
+      TID nIndex;
+      float size = 0;
+      float x, y, z;
+      float center[3] = {0, 0, 0};
+
+      while(stackSize > 0) {
+        cIndex = stack[--stackSize];
+
+        // update node data
+        triangulation->getVertexPoint(cIndex, x, y, z);
+        center[0] += x;
+        center[1] += y;
+        center[2] += z;
+        size++;
+
+        // add neihbors
+        size_t nNeighbors = triangulation->getVertexNeighborNumber(cIndex);
+        for(size_t i = 0; i < nNeighbors; i++) {
+          triangulation->getVertexNeighbor(cIndex, i, nIndex);
+          if(labels[nIndex] == this->UNLABELED) {
+            labels[nIndex] = componentId;
+            stack[stackSize++] = nIndex;
+          }
+        }
+      }
+      center[0] /= size;
+      center[1] /= size;
+      center[2] /= size;
+
+      // create component
+      components.resize(componentId + 1);
+      auto &c = components[componentId];
+      std::copy(center, center + 3, c.center);
+      c.size = size;
+      c.seed = seed;
+
+      return 1;
+    }
+
+    template <typename DT>
+    int initializeOutputLabels(int *labels,
+                               const TID nVertices,
+                               const DT *featureMask = nullptr) const {
+      if(featureMask) {
+        for(TID i = 0; i < nVertices; i++)
+          labels[i] = featureMask[i] > 0 ? this->UNLABELED : this->IGNORE;
+      } else {
+        for(TID i = 0; i < nVertices; i++)
+          labels[i] = this->UNLABELED;
+      }
+
+      return 1;
+    }
+
+    template <typename TT = ttk::AbstractTriangulation>
+    int computeConnectedComponents(std::vector<Component> &components,
+                                   int *outputLabels,
+                                   const TT *triangulation,
+                                   const bool useSeedAsComponentId
+                                   = false) const {
+
+      Timer timer;
+
+      TID nVertices = triangulation->getNumberOfVertices();
+
+      // init stack
+      std::vector<TID> stack(nVertices);
+
+      std::string msg = "Computing Connected Components";
+      this->printMsg(msg, 0, 0, 1, ttk::debug::LineMode::REPLACE);
+
+      for(TID i = 0; i < nVertices; i++)
+        if(outputLabels[i] == this->UNLABELED)
+          this->computeFloodFill<TT>(
+            outputLabels, components, stack, triangulation, i);
+
+      this->printMsg(msg, 1, timer.getElapsedTime(), 1);
+
+      if(useSeedAsComponentId) {
+        timer.reStart();
+        msg = "Labeling Components by Seed Id";
+        this->printMsg(msg, 0, 0, 1, ttk::debug::LineMode::REPLACE);
+
+        for(TID i = 0; i < nVertices; i++) {
```

Comment:
`for(auto cid : outputLables)` ?

Reply:
OutputLabels is a pointer and therefore automatic iteration is not possible.